### PR TITLE
Blade disc.sg

### PR DESCRIPTION
**Pull request by zhangxiao-stack**

### CODE REVIEW

File: tao_compiler/mlir/disc/transforms/disc_gpu_kernel_to_blob.cc [hunk 1]
```diff
@@ -280,7 +281,9 @@ class GpuKernelToBlobPass
     // Parse ROCm architecture.
     absl::string_view consumable_arch(arch_str);
     if (!absl::ConsumePrefix(&consumable_arch, "gfx")) {
-      return tensorflow::errors::Internal(
+//      return tensorflow::errors::Internal(
```
Comment (tanyokwok) [inline tao_compiler/mlir/disc/transforms/disc_gpu_kernel_to_blob.cc:284]: Nit: remove unused codes

File: tao_compiler/mlir/disc/transforms/disc_gpu_kernel_to_blob.cc [hunk 2]
```diff
@@ -311,14 +314,18 @@ class GpuKernelToBlobPass
           new llvm::raw_fd_ostream(ll_path, ec, llvm::sys::fs::OF_None));
       llvmModule->print(*ll_fs, nullptr);
       ll_fs->flush();
-      const auto& blob = hsaco_or.ValueOrDie();
+//      const auto& blob = hsaco_or.ValueOrDie();
```
Comment (tanyokwok) [inline tao_compiler/mlir/disc/transforms/disc_gpu_kernel_to_blob.cc:317]: ditto

File: tao_compiler/mlir/disc/transforms/disc_gpu_kernel_to_blob.cc [hunk 3]
```diff
@@ -311,14 +314,18 @@ class GpuKernelToBlobPass
...
+//add support for dcu
+      const auto& blob = hsaco_or.value();
       std::string llvmpath = kname + "_debugllvm.hsaco";
       auto myfile = std::fstream(llvmpath, std::ios::out | std::ios::binary);
       myfile.write((char*)blob.data(), blob.size());
       myfile.close();
       LOG(INFO) << "llvm hsaco to " << llvmpath;
       hipModule_t hipmodule;
-      hipError_t res = tensorflow::wrap::hipModuleLoadData(
+//    hipError_t res = tensorflow::wrap::hipModuleLoadData(
```
Comment (tanyokwok) [inline tao_compiler/mlir/disc/transforms/disc_gpu_kernel_to_blob.cc:326]: ditto

File: tao_compiler/mlir/xla/ral/BUILD [hunk 4]
```diff
@@ -551,7 +551,8 @@ disc_cc_library(
     hdrs = [
         "context/base/cuda/cuda_context_impl.h",
     ],
-    copts = ["-DTAO_RAL_USE_STREAM_EXECUTOR"],
+#   copts = ["-DTAO_RAL_USE_STREAM_EXECUTOR"],
```
Comment (tanyokwok) [inline tao_compiler/mlir/xla/ral/BUILD:554]: ditto

File: scripts/python/tao_build.py [hunk 5]
```diff
@@ -402,6 +403,9 @@ def test_tao_compiler(root, args):
     TARGET_DISC_CUDA_SOURCE_TESTS = [
         "//tensorflow/compiler/mlir/disc/tools/disc-source-emitter/tests/..."
     ]
+    TARGET_DISC_TRANSFORM_DIALECT_TESTS = [
```
Comment (tanyokwok) [inline scripts/python/tao_build.py:406]: This file changes too much, add @wyzero for review.

File: scripts/python/common_setup.py [hunk 6]
```diff
@@ -666,29 +666,102 @@ def get_version_file():
     root = get_source_root_dir()
     return os.path.join(root, "VERSION")
 
-if __name__ == "__main__":
-    parser = argparse.ArgumentParser()
+def add_arguments_common(parser):
```
Comment (tanyokwok) [inline scripts/python/common_setup.py:669]: This file changes too much, add @wyzero for review

**Comment by tanyokwok** [comment]
@zhangxiao-stack Nice job! Overall look good to me; I left some tiny comments.

File: scripts/python/tao_build.py (hunk 5 above)
Comment (wyzero) [reply scripts/python/tao_build.py:406]: Maybe we need to rebase to the main branch first? this part of code seems to be from the main branch.

**Pull request closed without merging**